virttest/qemu_devices/qcontainer.py

```diff
@@ -1714,10 +1714,33 @@ def define_hbas(qtype, atype, bus, unit, port, qbus, pci_bus, iothread,
                 access_secret, secret_type = sec, sectype
 
         iscsi_initiator = None
+        gluster_debug = None
+        gluster_logfile = None
+        gluster_peers = {}
         access = image_access.image_auth if image_access else None
         if access is not None:
             if access.storage_type == 'iscsi-direct':
                 iscsi_initiator = access.iscsi_initiator
+            elif access.storage_type == 'glusterfs-direct':
+                gluster_debug = access.debug
+                gluster_logfile = access.logfile
+
+                # it's always 'inet' for both ipv4 and ipv6 addresses
+                transport = 'inet' if not access.transport or access.transport == 'tcp' else access.transport
+                port = '0' if not access.port else access.port
```

Comment:
We don't have to set `port` explicitly, instead, can let qemu perform that.

Reply:
I am afraid we have to set it explicitly, or we will get the following error: 
Parameter 'port' is missing
If we set it to "", we will get the error:
'' is not a valid port number
So we have to set it here.